d2/src/main/java/com/linkedin/d2/xds/LoadBalanceTaskPool/NewLoadBalancerTaskthreadPool.java

```diff
@@ -0,0 +1,29 @@
+package com.linkedin.d2.xds.LoadBalanceTaskPool;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+public class NewLoadBalancerTaskthreadPool {
+  private ThreadPoolExecutor _threadPoolExecutor;
+  private int _corePoolSize = 2;
+  private int _maximumPoolSize = 3;
+  private long _keepAliveTime = 200;
+  private int _queueSize = 1000;
+
+  public NewLoadBalancerTaskthreadPool() {
+    RejectedExecutionHandler rejectedExecutionHandler = new NewBalanceTaskRejectedPolicy();
+    _threadPoolExecutor = new ThreadPoolExecutor(_corePoolSize, _maximumPoolSize, _keepAliveTime, TimeUnit.MILLISECONDS,
```

Comment:
It's usually necessary to use LinkedIn's executor services (for eg. the one created from `ScheduledExecutorServiceFactory` – although that one is quite old and I remember seeing some newer ones) since they provide IC handling and instrumentation for sending metrics to inGraphs, etc.

It'll need to be wired in from `container`, maybe via `D2ClientConfig`, as is done for other d2 executors.

Reply:
Thanks. That makes sense. Could I learn which `executor services` u mentiond ? As I see lot's of services will directly use `ThreadPool` to process tasks.

What I have done :
- use `D2ClientConfig` , which could be wired with `container`
- If there is no `ThreadPool` setted by `container`, the `DualReadLoadBalancer` will create a default `ThreadPool`